lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -1689,7 +1689,15 @@ def __init__(
         self._relation_name = relation_name
         self._container = self._get_container(container_name)
         self._container_name = self._get_container_name(container_name)
-        self._log_files = log_files or []
+
+        if log_files is None:
+            log_files = []
+        elif type(log_files) is not list:
+            raise TypeError(
+                f"The `log_files` argument must be a list but it is '{type(log_files)}'"
+            )
```

Comment:
This cannot be merged. Yes, we _should_ accept a list, and yes, accepting `Union[List[str], str]` may seem incoherent, but suddenly raising an exception in a library update which doesn't also increment `LIBAPI` with arguments which used to pass is an unacceptable workflow/regression, regardless of whether that behavior was intentional or not.

Hyrum's Law and so on. The fact that the user who reported this as not being enforced noticed _while_ doing this is a fair enough indication of this. Even while iterating over a string would end up with `/` as a log path, it remains.

`v1` probably isn't far away, at least.

Also, `if type(...)` is almost never the right choice over `isinstance`